hsds/dset_lib.py

```diff
@@ -0,0 +1,272 @@
+            dn_url = getDataNodeUrl(app, chunk_id)
+            req = dn_url + "/chunks/" + chunk_id
+            task = asyncio.ensure_future(http_delete(app, req, params=params))
+            tasks.append(task)
+        done, pending = await asyncio.wait(tasks)
+        if pending:
+            # should be empty since we didn't use return_when parameter
+            log.error("removeChunks - got pending tasks")
+            raise ValueError()
+        for task in done:
+            if task.exception():
+                exception_type = type(task.exception())
+                msg = f"removeChunks - task had exception: {exception_type}"
+                log.warn(msg)
+                failed_count += 1
+
+    except ClientError as ce:
+        msg = f"removeChunks - ClientError: {ce}"
+        log.error(msg)
+        raise ValueError()
+    except asyncio.CancelledError as cle:
+        log.error(f"removeChunks - CancelledError: {cle}")
+        raise ValueError()
+
+    if failed_count:
+        msg = f"removeChunks, failed count: {failed_count}"
+        log.error(msg)
+    else:
+        log.info(f"removeChunks complete for {len(chunk_ids)} chunks - no errors")
+
+
+async def getAllocatedChunkIds(app, dset_id, bucket=None):
+    """ Return the set of allocated chunk ids for the give dataset.
+        If slices is given, just return chunks that interesect with the slice region """
+
+    log.info(f"getAllocatedChunkIds for {dset_id}")
+
+    if not isSchema2Id(dset_id):
+        msg = f"no tabulation for schema v1 id: {dset_id} returning "
+        msg += "null results"
+        log.warn(msg)
+        return {}
+
+    if not bucket:
+        bucket = config.get("bucket_name")
+    if not bucket:
+        raise ValueError(f"no bucket defined for getAllocatedChunkIds for {dset_id}")
+
+    root_key = getS3Key(dset_id)
+    log.debug(f"got root_key: {root_key}")
+
+    if not root_key.endswith("/.dataset.json"):
+        raise ValueError("unexpected root key")
+
+    root_prefix = root_key[: -(len(".dataset.json"))]
+
+    log.debug(f"scanRoot - using prefix: {root_prefix}")
+
+    kwargs = {
+        "prefix": root_prefix,
+        "include_stats": False,
+        "bucket": bucket,
+    }
+    s3keys = await getStorKeys(app, **kwargs)
+
+    # getStoreKeys will pick up the dataset.json as well,
+    # so go through and discard
+    chunk_ids = []
+    for s3key in s3keys:
+        if s3key.endswith("json"):
+            # ignore metadata items
+            continue
+        try:
+            chunk_id = getObjId(s3key)
+        except ValueError:
+            log.warn(f"ignoring s3key: {s3key}")
+            continue
+        chunk_ids.append(chunk_id)
+
+    log.debug(f"getAllocattedChunkIds - got {len(chunk_ids)} ids")
+    return chunk_ids
+
+
+async def reduceShape(app, dset_json, shape_update, bucket=None):
+    """ Given an existing dataset and a new shape,
+        Reinitialize and edge chunks and delete any chunks
```

Comment:
Typo: `Reinitialize and edge chunks` -> `Reinitialize any edge chunks`

Reply:
updated